support/agent/miner_agent.go

```diff
@@ -480,6 +536,40 @@ func (ma *MinerAgent) scheduleSyncAndNextProof(v SimState, dlIdx uint64) error {
 	return nil
 }
 
+// Fill sector with deals
+// This is a naive packing algorithm that adds pieces in order received.
+func (ma *MinerAgent) fillSectorWithPendingDeals(expiration abi.ChainEpoch) ([]abi.DealID, abi.ChainEpoch) {
+	var dealIDs []abi.DealID
+
+	sectorSize, err := ma.Config.ProofType.SectorSize()
+	if err != nil {
+		panic(err)
+	}
+
+	// pieces are aligned so that each starts at the first multiple of its piece size >= the next empty slot.
+	// just stop when we find one that doesn't fit in the sector. Assume pieces can't have zero size
+	loc := uint64(0)
+	for _, piece := range ma.dealsPendingInclusion {
+		size := uint64(piece.size)
+		loc = (loc + size - 1) / size * size
```

Comment:
Another pair of parentheses would help greatly I spent a few minutes confused why we were dividing by the square of the size.